From linux-usb-owner@vger.kernel.org  Tue Oct  5 15:53:21 2010
From: Martin Fuzzey <mfuzzey@gmail.com>
Subject: USB: usbtest - ensure correct isoc data length
To: Greg KH <greg@kroah.com>, David Brownell <david-b@pacbell.net>,
	linux-usb@vger.kernel.org
Date: Fri, 01 Oct 2010 00:20:48 +0200
Message-ID: <20100930222048.15060.57008.stgit@srv002.fuzzey.net>

Check the data length of isochronous transfers is
as expected.

With this test #16 will now fail if the device side
sends no data.

Signed-off-by: Martin Fuzzey <mfuzzey@gmail.com>
Cc: David Brownell <david-b@pacbell.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/misc/usbtest.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/usb/misc/usbtest.c
+++ b/drivers/usb/misc/usbtest.c
@@ -1363,6 +1363,8 @@ static void iso_callback (struct urb *ur
 		ctx->errors += urb->error_count;
 	else if (urb->status != 0)
 		ctx->errors += urb->number_of_packets;
+	else if (urb->actual_length != urb->transfer_buffer_length)
+		ctx->errors++;
 
 	if (urb->status == 0 && ctx->count > (ctx->pending - 1)
 			&& !ctx->submit_error) {
